src/tools/hypothesis_testing.ts

```
@@ -35,6 +64,242 @@ function interpretCorrelation(r: number): string {
+  if (x.length !== y.length) {
+    throw new ValidationError('Correlation test requires series of equal length');
+  }
+  const n = x.length;
+  if (n < 3) {
+    throw new ValidationError(`Correlation test requires at least 3 observations, got ${n}`);
+  }
+  const r = toNumber(mathjs.corr(x, y));
+  const df = n - 2;
+  // As |r| -> 1 the statistic diverges; the p-value limit is 0.
+  const pValue =
+    Math.abs(r) >= 1 ? 0 : tTestPValue(r * Math.sqrt(df / (1 - r * r)), df, tail);
```

ESLint

[error] Delete `⏎···`

(prettier/prettier)